jembi/JeMPI
# Ensuring keycloak (and api_kc) use haproxy

**Ticket:** N/A
**Other Related Tickets:** N/A
**Other Related PRS:**  https://github.com/jembi/JeMPI/pull/132

----

## Describe of changes

This changes update the merged pull request https://github.com/jembi/JeMPI/pull/132 by ensuring that keycloak and the api use haproxy. 

**Please note**

With this change the ui app not required to login if you want to call api endpoints. This means that although setting the `REACT_APP_ENABLE_SSO`  to `false`  (see devops/linux/docker/conf/env/create-env-linux-low-1.sh ) will by pass the login screen, all other requests to the backed will fail. 

Backend services (such as the linker) use the non keycloak version of the api, and there is no need to authenticate there. This sits in its own docker network.  

## Login Details

This test server has one user with the details below

**Username:** dev-user

**Password:** dev-password

The admin use for the keycloak server is

**Username:** admin

**Password:** admin


## Future considerations

If none authentication use is still required for JeMPI, we can consider updating the haproxy config to route to the appropiate api based on the `REACT_APP_ENABLE_SSO` set

## How to test / configure

- Run JeMPI as you would normally 
- On accessing JeMPI UI you should see a login with keycloack screen
- Once you click login, you will be redirect to the keycloack login screen
- Enter the dec username and  password indicated above
- From there you should get logged in
